moveit_planners/ompl/ompl_interface/src/model_based_planning_context.cpp

```diff
@@ -512,7 +529,7 @@ bool ompl_interface::ModelBasedPlanningContext::solve(planning_interface::Motion
   if (solve(request_.allowed_planning_time, request_.num_planning_attempts))
   {
     double ptime = getLastPlanTime();
-    if (simplify_solutions_ && ptime < request_.allowed_planning_time)
+    if (simplify_solutions_)
```

Comment:
Another thing to note is that the usage of the path simplifier is slightly changed. The semantics are now "do at least one simplification pass, and then more if time allows" rather than "only if time allows". This is related to some [recent changes in OMPL](https://github.com/ompl/ompl/commit/da5f736058523485e087fd22a2b6524131fd1c40) that make the expected simplifier behavior explicit.

This is also a nice change for the asymptotically optimal planners like RRT*, as no simplification pass would be run as the planner used all available time budget. Now, a quick pass is done and a better path is generated (in general)